### ISSUE TITLE
Feat(eos_validate_state): ANTA Add a filter to target a specific set of devices

### ISSUE THREAD

**Issue by carl-baillargeon** [aristanetworks/ansible-avd#3305]
In CI pipelines, we can run jobs in parallel (matrix) to speed up validation. To make it work, a `device_filter` knob is needed to target a specific set of devices but still load the structured_config of all devices to avoid false positives and generate all required tests.

**Comment by carl-baillargeon** [comment]
Closing this for now. No longer needed/required since we optimized `eos_validate_state` ANTA mode a lot in recent development.

Can be reopened later if needed.
